docs/Reference/Task-Statuses.md

```diff
@@ -0,0 +1,309 @@
+# Task Statuses
+
+This document provides an overview of the various task statuses used in our
+system, along with their meanings, usage contexts, and the logic that determines
+these statuses.
+
+---
+
+## Table of Contents
+
+- [Active Task Statuses](#active-task-statuses)
+  - [`undispatched`](#undispatched)
+  - [`dispatched`](#dispatched)
+  - [`started`](#started)
+- [Final Task Statuses](#final-task-statuses)
+  - [`success`](#success)
+  - [`failed`](#failed)
+  - [`aborted`](#aborted)
+- [Display Status Logic](#display-status-logic)
+  - [Status Determination Logic](#status-determination-logic)
+  - [Display Statuses](#display-statuses)
+- [Task Icon Reference](#task-icon-reference)
+- [Notes](#notes)
+
+---
+
+## Active Task Statuses
+
+These statuses indicate tasks that are currently in progress or scheduled to
+run.
+
+### `undispatched`
+
+**Description**: Indicates one of the following:
+
+1. **Not Scheduled to Run**: The task is not activated
+   (`Task.Activated == false`).
+2. **Scheduled to Run**: The task is activated (`Task.Activated == true`) but
+   has not yet been dispatched to an agent.
+
+**Usage**: Reflects tasks awaiting dispatch, whether they're scheduled or not.
+
+---
+
+### `dispatched`
+
+**Description**: An [agent](./Glossary.md) has received the task, but the agent
+has not yet notified the system that it's running the task.
+
+**Usage**: Represents tasks that are assigned to agents but are not yet in
+execution.
+
+---
+
+### `started`
+
+**Description**: The task is currently running on an agent.
+
+**Usage**: Indicates active execution of a task.
+
+---
+
+## Final Task Statuses
+
+These statuses indicate that a task has completed its execution.
+
+### `success`
+
+**Description**: The task has finished successfully without any errors.
+
+**Usage**: Denotes tasks that have met all requirements and passed all tests.
+
+---
+
+### `failed`
+
+**Description**: The task has finished but encountered failures. This status
+covers any failure reason, which can be detailed in the task's end details.
+
+**Usage**: General failure status for tasks that did not complete successfully.
+
+---
+
+### `aborted`
```

Comment:
Aborted is actually not a valid DB status field, only a display status for tasks that have `t.Aborted == true`. The only values the status field can have in DB for a finished task are `success` or `failed`.